File: Sorting Algorithms/Counting Sort/countingSort.java
import java.util.Arrays;

public class countingSort{
    public static void CountingSort(int[] arr) {
        
        if (arr.length == 0) return;

        int maxVal = arr[0];
        for (int num: arr){
            if (num > maxVal){
                maxVal = num;
            }
        }

        int[] count = new int[maxVal + 1];

        for (int num: arr){
            count[num]++;
        }

        int index=0;
        for (int i=0; i < count.length; i++){
            while (count[i] > 0){
                arr[index] = i;
                index++;
                count[i]--;
            }
        }
    }

    public static void main(String[] args) {
        int[] myarr = {3, 2, 4, 5, 8, 11, 8, 45, 22, 65, 32, 87, 54, 99, 12, 34, 21, 56};
        CountingSort(myarr);
        System.err.println(Arrays.toString(myarr));
    }

}

File: Sorting Algorithms/Counting Sort/countingSort.py
def countingSort(arr:list) -> list:
    maxVal:int = max(arr)
    count:int = [0] * (maxVal + 1)
    
    while len(arr) > 0:
        num = arr.pop(0)
        count[num] += 1 
    
    for i in range(len(count)):
        while count[i] > 0: 
            arr.append(i)
            count[i] -= 1
    
    return arr

myarr:list = [3,2,4,5,8,11,8,45,22,65,32,87,54,99,12,34,21,56]
Sortarr:list = countingSort(myarr)
print(Sortarr)


